packages/relay-sdk/src/index.ts

```diff
@@ -0,0 +1,151 @@
+
+type SupportedNetworks = 'habanero' | 'manzano';
+
+function getRelayURLByNetwork(network: SupportedNetworks): string {
+    switch (network) {
+        case 'habanero':
+            return 'https://habanero-relayer.getlit.dev';
+        case 'manzano':
+            return 'https://manzano-relayer.getlit.dev';
+    }
+}
+
+export class LitRelayClient {
+
+    private readonly baseUrl: string;
+    private readonly apiKey: string;
+
+    private payerSecret: string | undefined = undefined;
+
+    /**
+     * Create a new LitRelayClient instance. Requires that the payer is already registered.
+     * and the the payer secret is known.
+     * 
+     * ```typescript
+     * const client = new LitRelayClient('https://habanero.lit.dev', 'my-api-key', 'my-payer-secret');
+     * ```
+     * 
+     * @param baseUrl
+     * @param apiKey
+     * @param payerSecret
+     */
+    private constructor(baseUrl: string, apiKey: string, payerSecret: string) {
+        this.baseUrl = baseUrl;
+        this.apiKey = apiKey;
+
+        if (payerSecret) {
+            this.payerSecret = payerSecret;
+        }
+    }
+
+    /**
+     * Adds a new payee to the payer's delegation list.
+     * 
+     * ```typescript
+     * const client = await LitRelayClient.connect('habanero', 'my-api-key', 'my-payer-secret');
+     * const result = await client.addPayee('payee-wallet-address');
+     * ```
+     * @param payeeAddress
+     * 
+     * @returns Promise<{
+     *  success: false;
+     *  error: string;
+     * } | {
+     *  success: true;
+     *  tokenId: string;
+     * }>
+     */
+    public async addPayee(payeeAddress: string) {
+        if (!this.payerSecret) {
+            throw new Error('Payer secret not set');
+        }
+
+        const res = await fetch(`${this.baseUrl}/add-users`, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                'api-key': this.apiKey,
+                'payer-secret-key': this.payerSecret,
+            },
+            body: JSON.stringify([payeeAddress]),
+        });
+
+        if (res.status !== 200) {
+            return {
+                success: false,
+                error: 'Failed to add payee: request failed',
+            };
+        }
+
+        const data = await res.json();
+
+        return {
+            success: true,
+            tokenId: data.tokenId,
+        };
```

Comment:
Hmm, the idea was to force the user to handle an error state (by throwing) rather than returns a failed promise that could be left dangling, but looking at this function again there is actually a lot of inconsistencies around it's return values. I'll try to figure out which result stratagy this module should follow.